torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -1745,3 +1745,76 @@ def rref_tensor_is_owner(rref_var):
 
         res = rref_tensor_is_owner(rref_var)
         self.assertEqual(res, False)
+
+    @dist_init
+    def test_rpc_async_op(self):
+        @torch.jit.script
+        def two_args_two_kwargs(
+            first_arg,
+            second_arg,
+            first_kwarg=torch.tensor([3, 3]),
+            second_kwarg=torch.tensor([4, 4]),
+        ):
+            return first_arg + second_arg + first_kwarg + second_kwarg
+
+        if self.rank != 0:
+            return
+
+        @torch.jit.script
+        def rpc_async_in_torchscript(dst_worker_name, args, kwargs):
+            # type: (str, Tuple[Tensor, Tensor], Dict[str, Tensor])
+            fut = rpc.api._invoke_rpc_torchscript(
+                dst_worker_name, two_args_two_kwargs, args, kwargs
```

Comment:
Shouldn't we test `_rpc_async_torchscript` which is the user facing API instead of `_invoke_rpc_torchscript`? Also, how does specifying `two_args_two_kwargs` work here? Doesn't `_invoke_rpc_torchscript` take a string as a qualified name?

Reply:
> Shouldn't we test _rpc_async_torchscript which is the user facing API instead of _invoke_rpc_torchscript?

`_rpc_async_torchscript(..)` is not scriptable. 
Notice it passes `user_func` to `_invoke_rpc_torchscript(..)`.
For JIT interpreter, all values being passed around are `IValue`s. 
While `IValue` does not support function type, so it can't hold a Python function `user_func`.

We need to have JIT support function type IValue before doing this?

> Shouldn't we test _rpc_async_torchscript which is the user facing API instead of _invoke_rpc_torchscript? Doesn't _invoke_rpc_torchscript take a string as a qualified name?

I have changed it to take a user function instead. 
JIT IR emitter must know the function return type. This is for telling what type the async rpc returned future contains.

Function qualified name does not have this return type information. 
And, also function contains function qualified name, so it could be passed to `_invoke_rpc_torchscript(..)`.

Reply:
We should make `_rpc_async_torchscript` torchscriptable (by removing user_func if necessary). `_rpc_async_torchscript` is the API we want users to use and `_invoke_rpc_torchscript` should be an internal API. In the future, we probably would make `_rpc_async_torchscript` a public API for users to use.

Reply:
@pritamdamania87 

As @zhaojuanmao asked above, we are removing `_rpc_async_torchscript` and _rpc_sync_torchscript`, the public API will be the same for both languages.